webapp/src/pages/Settings.tsx

```diff
@@ -746,7 +747,65 @@ const Settings: React.FC<Props> = ({ onClose }) => {
           ) : Array.isArray(value) ? (
             displayArgumentsList(field, value)
           ) : typeof value === "object" ? (
-            displayKeywordArguments(field, value)
+            <Box
+              key={field}
+              display="flex"
+              flexDirection="column"
+              marginRight={2}
+            >
+              <Typography variant="caption">{field}</Typography>
+              <KeyValuePairs>
+                {Object.entries(value).map(([objField, objValue], index) => (
+                  <React.Fragment key={index}>
+                    <Typography variant="body2">{objField}:</Typography>
+                    {Array.isArray(objValue) ? (
+                      <Tooltip title={objValue.join(", ")}>
+                        <StringField
+                          key={objField}
+                          value={objValue.join(", ")}
+                          disabled={
+                            isUpdatingConfig ||
+                            (config && !resultingConfig[config])
+                          }
+                          onChange={(newValue) =>
+                            setPartialConfig({
+                              ...partialConfig,
+                              [config]: {
+                                ...resultingConfig[config],
+                                [field]: {
+                                  ...value,
+                                  [objField]: newValue.split(", "),
+                                },
+                              },
+                            })
+                          }
+                        />
+                      </Tooltip>
+                    ) : (
+                      <NumberField
+                        sx={{ width: "10ch" }}
+                        key={index}
+                        value={objValue as number}
+                        disabled={!resultingConfig[config] || isUpdatingConfig}
+                        onChange={(newValue) =>
+                          setPartialConfig({
+                            ...partialConfig,
+                            [config]: {
+                              ...resultingConfig[config],
+                              [field]: {
+                                ...value,
+                                [objField]: newValue,
+                              },
+                            },
+                          })
+                        }
+                        {...FIELDS[field]}
```

Comment:
I think you meant
```suggestion
                        {...FIELDS[objField]}
```
since `bevavioral_testing` doesn't exist on `FIELDS`. However, the `threshold` fields will conflict with the existing `FIELDS.threshold`, so I think I would simply do
```suggestion
                        {...INT}
```

Reply:
You can now remove `nb_typos_per_utterance` from `FIELDS`.

Reply:
Same